### PR TITLE
Add name and location query

### PR DESCRIPTION
**Pull request by penrods**
This add the ability to ask the name and the location settings
for the device.  Also switches to the Apache 2.0 license and
updates the README with more info.

### CODE REVIEW

**Comment by MatthewScholefield** [comment]
This will fail on older devices without intent decorators or the configuration change. If breaking compatibility is intended, then make sure to also remove the unnecessary try catch in precise methods, and add the intent decorators to the other handlers.

**Comment by penrods** [comment]
We added intent_hander() 10 months ago.  If they still haven't updated, then they need to.  :)

We officially support 17.08 and forward, which is 0.9+.  The localConf  was introduced 3 months ago, so that is outside the 0.9+ window, but since some of the early shipping units predated this (June-ish) it is probably worth keeping the backwards compatibility.

**Comment by MatthewScholefield** [comment]
That makes sense. If that's what we are deciding to support, all I see that needs to be changed is:
 - Change remaining handlers to decorator
 - Import legacy `ConfigurationManager` instead of `Configuration`
 - Remove try catch on handle_update_precise and handle_get_listener (since `ConfigurationManager.get()` is always safe to use)

**Comment by penrods** [comment]
Ping

File: __init__.py [hunk 1]
```diff
@@ -36,29 +34,33 @@ def __init__(self):
...
+    @intent_handler(IntentBuilder('').require("What").require("Name"))
+    def handle_query_name(self, message):
+        device = DeviceApi().get()
+        self.speak_dialog("my.name.is", data={"name": device["name"]})
+
+    @intent_handler(IntentBuilder('').require("What").require("Location"))
+    def handle_what_is_location(self, message):
+        # "what is your location" is the same as "where are you", but
+        # was difficult to fit into the same intent vocabulary
+        self.where_are_you(message)
```
Comment (MatthewScholefield) [inline __init__.py]:
```python
self.handle_where_are_you
```

**Review by MatthewScholefield — CHANGES_REQUESTED** [review]
Looks good. Just a small typo to fix.

**Review by MatthewScholefield — APPROVED** [review]

**Pull request merged**